### PR TITLE
Fix/ar meta up to date

### PR DESCRIPTION
**Pull request by paulmskim**
For tickets:
- https://central.tri.be/issues/121592
- https://central.tri.be/issues/121819
- https://central.tri.be/issues/121821
- https://central.tri.be/issues/120735

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/dd0226119c9631e1ca11a8579693e5881a24cab9/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/89843004f4b080e8a4b6ccf32083db2bcb9ef76c/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

File: src/views/registration/attendees/success.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+/**
+ * This template renders the success message for each form
+ * of the attendee registration page
+ *
+ * @version 4.9
```
Comment (sc0ttkclark) [inline src/views/registration/attendees/success.php]:
```suggestion
 * @since TBD
 * @version TBD
```

File: src/Tribe/Tickets.php [hunk 2]
```diff
@@ -2416,31 +2416,40 @@ public function maybe_redirect_to_attendees_registration_screen( $redirect = nul
 				return;
 			}
 
+			$is_paypal              = ! ! $redirect;
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets.php]: Whoa, is this intentional? There's also a `if ( ! $is_paypal )` line below at 2426

File: src/Tribe/Tickets.php [hunk 3]
```diff
@@ -2416,31 +2416,40 @@ public function maybe_redirect_to_attendees_registration_screen( $redirect = nul
...
+				if (
+					$up_to_date
+						&& ! isset( $_REQUEST[ 'wootickets_process' ] )
+						&& ! isset( $_REQUEST[ 'eddtickets_process' ] )
+				) {
+					return;
+				}
+
+				// Bail if processing checkout for WooCommerce
+				if ( isset( $_REQUEST[ 'key' ] ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets.php]:
```suggestion
				if ( isset( $_REQUEST['key'] ) ) {
```

File: src/Tribe/Tickets.php [hunk 4]
```diff
@@ -2416,31 +2416,40 @@ public function maybe_redirect_to_attendees_registration_screen( $redirect = nul
...
-					! isset( $_REQUEST['wootickets_process'] )
-					&& ! isset( $_REQUEST['eddtickets_process'] )
-				)
-			) {
+				// Bail If things are up to date and they haven't submitted the form
+				// to access the registration page.
+				if (
+					$up_to_date
+						&& ! isset( $_REQUEST[ 'wootickets_process' ] )
+						&& ! isset( $_REQUEST[ 'eddtickets_process' ] )
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets.php]:
```suggestion
						&& ! isset( $_REQUEST['eddtickets_process'] )
```

File: src/Tribe/Tickets.php [hunk 5]
```diff
@@ -2416,31 +2416,40 @@ public function maybe_redirect_to_attendees_registration_screen( $redirect = nul
...
-				&& (
-					! isset( $_REQUEST['wootickets_process'] )
-					&& ! isset( $_REQUEST['eddtickets_process'] )
-				)
-			) {
+				// Bail If things are up to date and they haven't submitted the form
+				// to access the registration page.
+				if (
+					$up_to_date
+						&& ! isset( $_REQUEST[ 'wootickets_process' ] )
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets.php]:
```suggestion
						&& ! isset( $_REQUEST['wootickets_process'] )
```

File: src/Tribe/Tickets.php [hunk 6]
```diff
@@ -2416,31 +2416,40 @@ public function maybe_redirect_to_attendees_registration_screen( $redirect = nul
...
-				)
-			) {
-				return;
-			}
+			// If WooCommerce or EDD
+			if ( ! $is_paypal ) {
+				// Bail if there are no required fields in cart or the stored data is up to date
+				// And they're submitting the Attendee Registration page
+				if (
+					isset( $_REQUEST[ 'tribe_tickets_checkout' ] )
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets.php]:
```suggestion
					isset( $_REQUEST['tribe_tickets_checkout'] )
```

File: src/Tribe/Tickets.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/Tribe/Tickets.php]: !!? and then a ! check later?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/63a9ddd978b311b13fc3cbb66e119ba66f7ad656/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets.php (hunk 2 above)
Comment (paulmskim) [reply src/Tribe/Tickets.php]: Should I just `(bool)` cast it instead? the only two possible values I'm seeing are a `string` or `null`.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/d40d6a3f9bcaec22b57b4106777efeb18ecb54a0/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets.php (hunk 2 above)
Comment (paulmskim) [reply src/Tribe/Tickets.php]: I just did it.

**Comment by sc0ttkclark** [comment]
Can you resolve conflicts? Merge latest release/G19.03 branch into this and rebuild CSS

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/3702a40b2675db57ee8539a5e8117a57a072c9a2/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

**Comment by paulmskim** [comment]
This needs to be merged with https://github.com/moderntribe/event-tickets/pull/1076, waiting on confirmation from Gustavo and Crisoforo.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/7a26dcf1497afa2740ad332670a9f76ad3e56ecb/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

**Comment by sc0ttkclark** [comment]
Can you fix conflicts on this and let me know if everything in #1076 is all set for this to go out?

File: src/resources/js/tickets-registration-page.js [hunk 7]
```diff
@@ -92,27 +104,94 @@ tribe.tickets.registration = {};
...
+			$fields.find( obj.selector.fieldsErrorRequired ).show();
+			obj.updateStatusToIncomplete( $event )
+		} else {
+			$fields.find( obj.selector.loader ).show();
+
+			var eventId = $event.data( 'event-id' );
+			var params = $form.serializeArray();
+			params.push( { name: 'event_id', value: eventId } );
+			params.push( { name: 'action', value: 'tribe-tickets-save-attendee-info' } );
+			params.push( { name: 'nonce', value: TribeTicketsPlus.save_attendee_info_nonce } );
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: @paulmskim This references ET+ code, is this properly checking the var is available for use before trying to run this?

File: src/resources/js/tickets-registration-page.js [hunk 8]
```diff
@@ -92,27 +104,94 @@ tribe.tickets.registration = {};
...
+
+			$.post(
+				TribeTicketsPlus.ajaxurl,
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: @paulmskim This references ET+ code, is this properly checking the var is available for use before trying to run this?

File: src/resources/js/tickets-registration-page.js (hunk 7 above)
Comment (paulmskim) [reply src/resources/js/tickets-registration-page.js]: This is run on the attendee registration page. If you don't have ET+ you won't be able to access the attendee registration page and run this code. I can still add a check for this if necessary.

File: src/resources/js/tickets-registration-page.js (hunk 8 above)
Comment (paulmskim) [reply src/resources/js/tickets-registration-page.js]: Same note as above

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/6be69b0a87879e0c5838092df69b9c604b4561f2/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

**Comment by paulmskim** [comment]
Everything in https://github.com/moderntribe/event-tickets/pull/1076 is ready to go.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [View.php:60](https://github.com/moderntribe/event-tickets/blob/00fb2ba1d0830d2a64bc9d7b71f9539c845c8532/src/Tribe/Attendee_Registration/View.php#L60)

> via jenkins/codesniffer-phpcs

**Pull request merged**